Fullscreen status for all windows is now saved.

bsclifton opened the pull request:
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Ran `git rebase -i` to squash commits if needed.

Fullscreen status for all windows is now saved.
Fixes https://github.com/brave/browser-laptop/issues/2135

bsclifton commented:
Ready for review :smile: I did some manual testing with several windows and it worked great; would be awesome to try it on Ubuntu and Mac

luixxiul commented:
I tried it on Ubuntu 14.04 with 2 windows and it worked fine :-)

luixxiul commented:
On Ubuntu 14.04 if one window is fullscreen and another is not, restarting does not restore the fullscreen status of the first one.

bridiver commented on js/actions/windowActions.js:
```diff
@@ -738,6 +738,17 @@ const windowActions = {
+   * Sets the fullscreen state of the window
+   * @param {boolean} isFullScreen - true if window is fullscreen
+   */
+  setFullscreenState: function (isFullScreen) {
```
I think we need to do something to better differentiate webview and window fullscreen. The webview fullscreen state is triggered by the electron `enter-html-full-screen` event and windowActions.setFullScreen updates the `isFullScreen` property of the frame. Maybe we could have `setWebviewFullScreen' and `setWindowFullScreen` to make it more clear?

bridiver replied:
actually `setFrameFullScreen` is probably more consistent with other webview actions

bsclifton commented:
Updated per your feedback, @bridiver :smile: 

I also fixed a nasty unintentional behavior I introduced w/ capturing the window position. It was creating hundreds of window actions per second if you're moving the window. Fix was to add a sliding one second timer to the event. Much better (and noticeably faster)

bbondy commented:
Thanks, merging!

bridiver commented on js/components/main.js:
```diff
@@ -408,14 +408,31 @@ class Main extends ImmutableComponent {
+    let moveTimeout = null
     win.on('move', function (event) {
-      windowActions.savePosition(event.sender.getPosition())
+      if (moveTimeout) {
```
fyi - there is a `debounce` method that we also use in other places

bridiver commented:
++
